tests/DatabaseConnectionTest.php

```diff
@@ -0,0 +1,205 @@
+        $events->shouldReceive('fire')->once()->with(m::type('Illuminate\Database\Events\TransactionCommitted'));
+        $connection->commit();
+    }
+
+    public function testRollBackedFiresEventsIfSet()
+    {
+        $connection = $this->getMockConnection(['getName', 'db2rollback']);
+        $connection->expects($this->any())->method('getName')->will($this->returnValue('name'));
+        $connection->expects($this->once())->method('db2rollback');
+        $connection->setEventDispatcher($events = m::mock('Illuminate\Contracts\Events\Dispatcher'));
+        $events->shouldReceive('fire')->once()->with(m::type('Illuminate\Database\Events\TransactionRolledBack'));
+        $connection->rollBack();
+    }
+
+    public function testFromCreatesNewQueryBuilder()
+    {
+        $conn = $this->getMockConnection();
+        $conn->setQueryGrammar(m::mock('Illuminate\Database\Query\Grammars\Grammar'));
+        $conn->setPostProcessor(m::mock('Illuminate\Database\Query\Processors\Processor'));
+        $builder = $conn->table('users');
+        $this->assertInstanceOf('Illuminate\Database\Query\Builder', $builder);
+        $this->assertEquals('users', $builder->from);
+    }
+
+    public function testPrepareBindings()
+    {
+        $date = m::mock('DateTime');
+        $date->shouldReceive('format')->once()->with('foo')->andReturn('bar');
+        $bindings = ['test' => $date];
+        $conn = $this->getMockConnection();
+        $grammar = m::mock('Illuminate\Database\Query\Grammars\Grammar');
+        $grammar->shouldReceive('getDateFormat')->once()->andReturn('foo');
+        $conn->setQueryGrammar($grammar);
+        $result = $conn->prepareBindings($bindings);
+        $this->assertEquals(['test' => 'bar'], $result);
+    }
+
+    public function testLogQueryFiresEventsIfSet()
+    {
+        $connection = $this->getMockConnection();
+        $connection->logQuery('foo', [], time());
+        $connection->setEventDispatcher($events = m::mock('Illuminate\Contracts\Events\Dispatcher'));
+        $events->shouldReceive('fire')->once()->with(m::type('Illuminate\Database\Events\QueryExecuted'));
+        $connection->logQuery('foo', [], null);
+    }
+
+//    public function testPretendOnlyLogsQueries()
+//    {
+//        $connection = $this->getMockConnection();
+//        $queries = $connection->pretend(function ($connection) {
+//            $connection->select('foo bar', ['baz']);
+//        });
+//        $this->assertEquals('foo bar', $queries[0]['query']);
+//        $this->assertEquals(['baz'], $queries[0]['bindings']);
+//    }
+
+    public function testSchemaBuilderCanBeCreated()
+    {
+        $connection = $this->getMockConnection();
+        $schema = $connection->getSchemaBuilder();
+        $this->assertInstanceOf('Illuminate\Database\Schema\Builder', $schema);
+        $this->assertSame($connection, $schema->getConnection());
+    }
+
+//    public function testAlternateFetchModes()
+//    {
```

Comment:
see how different fetch methods are used in other connections & see if we need to implement this.